osrf/rmf_core
# Loop and Delivery integration with dispatcher framework

This PR adds support for loop and delivery tasks within the dispatcher framework. 

Can be tested with `feature/task_planner_params` branch of `rmf_demos`.
To test:
Launch office world.
Publish bid notice for Loop:
```
ros2 topic pub  /rmf_task/bid_notice rmf_task_msgs/BidNotice "{task_profile: {task_id: '1',task_type: {type: 1}, loop: {start_name: 'coe', finish_name: 'pantry', num_loops: 2} }}" --once
```

Award bid
```
ros2 topic pub  /rmf_task/dispatch_request rmf_task_msgs/DispatchRequest "{fleet_name: "tinyRobot", method: 1, task_profile: {task_id: '1', task_type: {type: 1} }}" --once
```

Publish bid notice for Delivery:
```
ros2 topic pub  /rmf_task/bid_notice rmf_task_msgs/BidNotice "{task_profile: {task_id: '2',task_type: {type: 2}, delivery: {pickup_place_name: 'pantry', pickup_dispenser: 'coke_dispenser', dropoff_place_name: 'hardware_2', dropoff_ingestor: 'coke_ingestor'} }}" --once
```

Award bid:
```
ros2 topic pub  /rmf_task/dispatch_request rmf_task_msgs/DispatchRequest "{fleet_name: "tinyRobot", method: 1, task_profile: {task_id: '2', task_type: {type: 2} }}" --once
```
